components/script/dom/readablestreamdefaultreader.rs

```diff
@@ -176,43 +176,93 @@ impl ReadableStreamDefaultReader {
             .pop_front()
             .expect("Reader must have read request when remove is called into.")
     }
+
+    /// <https://streams.spec.whatwg.org/#abstract-opdef-readablestreamdefaultreadererrorreadrequests>
+    pub fn empty_read_requests(&self) {
+        // step 2
+        self.read_requests.borrow_mut().clear();
+    }
 }
 
 impl ReadableStreamDefaultReaderMethods for ReadableStreamDefaultReader {
     /// <https://streams.spec.whatwg.org/#default-reader-read>
     fn Read(&self) -> Rc<Promise> {
         let promise = Promise::new(&self.reflector_.global());
 
-        self.stream
-            .perform_pull_steps(ReadRequest::Read(promise.clone()));
+        if let Some(stream) = self.stream.get() {
+            stream.perform_pull_steps(ReadRequest::Read(promise.clone()));
+        } else {
+            promise.reject_error(Error::Type("stream is undefined".to_owned()));
+        }
 
         promise
     }
 
     /// <https://streams.spec.whatwg.org/#default-reader-release-lock>
     fn ReleaseLock(&self) {
-        if self.stream.is_readable() {
-            self.closed_promise.reject_native(&());
+        // step 1
+        if self.stream.get().is_none() {
+            return;
+        }
+
+        // <https://streams.spec.whatwg.org/#readable-stream-reader-generic-release>
```

Comment:
let's put this in its own method, it is used in many other places in the spec, and also it's clearer with the spec link. We can do the same with other spec functions.